notify/factory_test.go

```diff
@@ -72,6 +76,56 @@ func TestBuildReceiverIntegrations(t *testing.T) {
 		t.Run("should call notify wrapper for each config", func(t *testing.T) {
 			require.Equal(t, qty, wrapped)
 		})
+		t.Run("should use custom dial context", func(t *testing.T) {
+			customDialError := fmt.Errorf("custom dial function error")
+			clientOpts := []http.ClientOption{
+				http.WithUserAgent("Grafana-test"),
+				http.WithDialer(net.Dialer{
+					// Also override the Resolver so that configuration that doesn't use real hostnamesalso return
+					// "custom dial function error" instead of "no such dns".
```

Comment:
Super nit, there's a space missing.
```suggestion
					// Also override the Resolver so that configuration that doesn't use real hostnames
					// also return "custom dial function error" instead of "no such dns".
```